File: app/src/main/java/com/github/mbarrben/moviedb/movies/MoviesActivity.java
package com.github.mbarrben.moviedb.movies;

import android.os.Bundle;
import com.github.mbarrben.moviedb.BaseActivity;
import com.github.mbarrben.moviedb.R;
import java.util.LinkedList;
import java.util.List;

public class MoviesActivity extends BaseActivity {

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.movies_activity);
  }

  @Override
  protected List<Object> getModules() {
    LinkedList<Object> modules = new LinkedList<>();
    modules.add(new MoviesListModule());
    return modules;
  }
}

File: app/src/main/kotlin/com/github/mbarrben/moviedb/movies/view/platform/MoviesActivity.kt
package com.github.mbarrben.moviedb.movies.view.platform

import com.github.mbarrben.moviedb.BaseActivity
import com.github.mbarrben.moviedb.R.layout
import com.github.mbarrben.moviedb.di.HasComponent
import com.github.mbarrben.moviedb.movies.di.DaggerMoviesListComponent
import com.github.mbarrben.moviedb.movies.di.MoviesListComponent
import com.github.mbarrben.moviedb.movies.di.MoviesListModule

class MoviesActivity : BaseActivity(layout.movies_activity), HasComponent<MoviesListComponent> {

  override val component: MoviesListComponent by lazy {
    DaggerMoviesListComponent.builder()
        .applicationComponent(applicationComponent)
        .activityModule(activityModule)
        .moviesListModule(MoviesListModule())
        .build()
  }

  override fun inject() = component.inject(this)

}
